mm2src/coins/lp_coins.rs

```diff
@@ -3595,29 +3614,142 @@ impl MmCoinStruct {
+    /// Dex fee with the burn amount
     WithBurn {
+        /// Amount to go to the dex fee address
         fee_amount: MmNumber,
+        /// Amount to be burned
         burn_amount: MmNumber,
+        /// This indicates how to burn the burn_amount
+        burn_destination: DexFeeBurnDestination,
     },
 }
 
 impl DexFee {
-    /// Creates a new `DexFee` with burn amounts.
-    pub fn with_burn(fee_amount: MmNumber, burn_amount: MmNumber) -> DexFee {
-        DexFee::WithBurn {
-            fee_amount,
-            burn_amount,
+    const DEX_FEE_SHARE: &str = "0.75";
+
+    /// Recreates a `DexFee` from separate fields (usually stored in db).
+    pub fn create_from_fields(fee_amount: MmNumber, burn_amount: MmNumber, ticker: &str) -> DexFee {
+        if burn_amount > MmNumber::default() {
+            let burn_destination = match ticker {
+                "KMD" => DexFeeBurnDestination::KmdOpReturn,
+                _ => DexFeeBurnDestination::PreBurnAccount,
+            };
+            DexFee::WithBurn {
+                fee_amount,
+                burn_amount,
+                burn_destination,
+            }
+        } else {
+            DexFee::Standard(fee_amount)
+        }
+    }
+
+    /// Creates a new `DexFee` for a taker coin to sell.
+    pub fn new_from_taker_coin(taker_coin: &dyn MmCoin, rel_ticker: &str, trade_amount: &MmNumber) -> DexFee {
+        // calc dex fee
+        let rate = Self::dex_fee_rate(taker_coin.ticker(), rel_ticker);
+        let dex_fee = trade_amount * &rate;
+        let min_tx_amount = MmNumber::from(taker_coin.min_tx_amount());
+        if dex_fee <= min_tx_amount {
+            return DexFee::Standard(min_tx_amount);
+        }
+
+        if taker_coin.should_burn_dex_fee() {
+            if taker_coin.is_kmd() {
+                // use a special dex fee option for kmd
+                let (fee_amount, burn_amount) = Self::calc_burn_amount_for_op_return(&dex_fee, &min_tx_amount);
+                return DexFee::WithBurn {
+                    fee_amount,
+                    burn_amount,
+                    burn_destination: DexFeeBurnDestination::KmdOpReturn,
+                };
+            } else {
+                // burn dex fee to the burn account
+                let (fee_amount, burn_amount) = Self::calc_burn_amount_for_burn_account(&dex_fee, &min_tx_amount);
+                // burn_amount can be set to zero if it is dust
+                if burn_amount > MmNumber::from(0) {
+                    return DexFee::WithBurn {
+                        fee_amount,
+                        burn_amount,
+                        burn_destination: DexFeeBurnDestination::PreBurnAccount,
+                    };
+                }
+            }
+        }
+        DexFee::Standard(dex_fee)
+    }
+
+    /// Returns dex fee discount if KMD is traded
+    pub fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
+        let fee_discount_tickers: &[&str] = match var("MYCOIN_FEE_DISCOUNT") {
```

Comment:
`MYCOIN_FEE_DISCOUNT` is only used in tests, how about this

```diff
Subject: [PATCH] add Omer Yacine to authors
---
Index: mm2src/coins/lp_coins.rs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mm2src/coins/lp_coins.rs b/mm2src/coins/lp_coins.rs
--- a/mm2src/coins/lp_coins.rs	(revision 06e36b50ba72c5d0d2625666dfa79582f1749032)
+++ b/mm2src/coins/lp_coins.rs	(date 1723079152850)
@@ -78,7 +78,6 @@
 use std::cmp::Ordering;
 use std::collections::hash_map::{HashMap, RawEntryMut};
 use std::collections::HashSet;
-use std::env::var;
 use std::future::Future as Future03;
 use std::num::{NonZeroUsize, TryFromIntError};
 use std::ops::{Add, AddAssign, Deref};
@@ -3696,10 +3695,15 @@
 
     /// Returns dex fee discount if KMD is traded
     pub fn dex_fee_rate(base: &str, rel: &str) -> MmNumber {
-        let fee_discount_tickers: &[&str] = match var("MYCOIN_FEE_DISCOUNT") {
+        #[cfg(test)]
+        let fee_discount_tickers: &[&str] = match std::env::var("MYCOIN_FEE_DISCOUNT") {
             Ok(_) => &["KMD", "MYCOIN"],
             Err(_) => &["KMD"],
         };
+
+        #[cfg(not(test))]
+        let fee_discount_tickers: &[&str] = &["KMD"];
+
         if fee_discount_tickers.contains(&base) || fee_discount_tickers.contains(&rel) {
             // 1/777 - 10%
             BigRational::new(9.into(), 7770.into()).into()
```

Reply:
fixed 217a313bc9439a6f6930d7b24fef3a28ce4bf16d